components/suggest/src/schema.rs

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+use rusqlite::{Connection, Transaction};
+use sql_support::open_database::{self, ConnectionInitializer};
+
+pub const VERSION: u32 = 1;
+
+pub const SQL: &str = "
+    CREATE TABLE meta(
+        key TEXT PRIMARY KEY,
+        value NOT NULL
+    ) WITHOUT ROWID;
+
+    CREATE TABLE keywords(
+        keyword TEXT NOT NULL,
+        suggestion_id INTEGER NOT NULL REFERENCES suggestions(id) ON DELETE CASCADE,
+        rank INTEGER NOT NULL,
+        PRIMARY KEY (keyword, suggestion_id)
+    ) WITHOUT ROWID;
+
+    CREATE UNIQUE INDEX keywords_suggestion_id_rank ON keywords(suggestion_id, rank);
+
+    CREATE TABLE suggestions(
+        id INTEGER PRIMARY KEY,
+        record_id TEXT NOT NULL,
+        block_id INTEGER NOT NULL,
+        advertiser TEXT NOT NULL,
```

Comment:
Good idea!

Going a little further—since we're not actually querying by `block_id`, `impression_url`, `click_url`, or `iab_category` (or, for that matter, by any of the other columns in `suggestions`, except for `id`, `record_id`, and maybe `provider`/`advertiser`)...we could make `custom_details` be a regular `TEXT` column in the `suggestions` table, and just store a big ol' JSON blob with all those fields, instead of unpacking them into a separate table!

(If we really wanted to query on those fields, SQLite also supports [JSON natively now](https://www.sqlite.org/json1.html), so that would be an option. It wouldn't be as efficient as normalizing them, and it might increase the database size a bit—so we may want to compare using JSON vs. separate tables for the custom details).

@tiftran, maybe something to think about when you work on this (and thanks for taking it on! 🚀)

Reply:
I filed https://mozilla-hub.atlassian.net/browse/DISCO-2537 for this work.

Reply:
Yup, those are dumb fields that we'd never need a secondary index on any of them. We can pack them up into a single text blob for now.